core/jobs/batch_jobs/exploration_validation_jobs.py

```diff
@@ -0,0 +1,154 @@
+# coding: utf-8
+#
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Validation jobs for exploration models."""
+
+from __future__ import annotations
+
+from core.constants import constants
+from core.domain import exp_fetchers
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.transforms import job_result_transforms
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+
+(exp_models, ) = models.Registry.import_models([models.NAMES.exploration])
+
+
+class GetExpWithInvalidCategoryJob(base_jobs.JobBase):
+    """Job that returns explorations with categories not in constants.ts"""
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        """Returns PCollection of invalid explorations with their id and
+        category.
+
+        Returns:
+            PCollection. Returns PCollection of invalid explorations with
+            their id and category.
+        """
+        total_explorations = (
+            self.pipeline
+            | 'Get all ExplorationModels' >> ndb_io.GetModels(
+                exp_models.ExpSummaryModel.get_all(include_deleted=False))
+            | 'Get exploration from model' >> beam.Map(
+                exp_fetchers.get_exploration_summary_from_model)
+        )
+
+        published_explorations = (
+            total_explorations
+            | 'Get published explorations' >> beam.Filter(
+                lambda exp: exp.first_published_msec is not None
```

Comment:
This is not a good filter. Please use the `status` field.

Reply:
Done!